### ISSUE TITLE
Crash in resolver.py: 'ascii' codec can't encode character u'\xfc' in position 32: ordinal not in range(128)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jeevandongre/namebench#40]
```
What steps will reproduce the problem?

1. Double Click on the downloaded .exe

What is the expected output? What do you see instead?

I would expect the program to install or load or something along those lines. 
I get an error message saying "See the logfile
C:\Users\Alex\AppData\Local\Temp\ckz_CNSF\namebench.exe.log"

What version of the product are you using? On what operating system?

I'm trying to use namebench-1.0.exe on Windows 7 64bit

Please provide any additional information below.
This is what is in the file from the error:

Traceback (most recent call last):
  File "namebench.py", line 74, in <module>
  File "libnamebench\tk.pyc", line 90, in Execute
  File "libnamebench\tk.pyc", line 122, in DrawWindow
  File "libnamebench\util.pyc", line 69, in InternalNameServers
  File "third_party\dns\resolver.pyc", line 293, in __init__
  File "third_party\dns\resolver.pyc", line 459, in read_registry
  File "third_party\dns\resolver.pyc", line 410, in _config_win32_fromkey
  File "third_party\dns\resolver.pyc", line 368, in _config_win32_nameservers
UnicodeEncodeError: 'ascii' codec can't encode character u'\xfc' in
position 32: ordinal not in range(128)


```

Original issue reported on code.google.com by `alexwie...@gmail.com` on 7 Dec 2009 at 11:51

**Comment by GoogleCodeExporter** [comment]
```
It's a bit bizarre to be honest. The crash is in the dnspython library, which 
is unable to 
read your registry for some reason. Somewhere there is a ü character. I'll put 
in a 
workaround so that if it cannot read the registry, that it doesn't crash.

```

Original comment by `tstromb...@google.com` on 8 Dec 2009 at 12:17
- Changed title: **Crash in resolver.py: 'ascii' codec can't encode character u'\xfc' in position 32: ordinal not in range(128)**
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Workaround committed in revision 414. Will be part of 1.0.1 release.


```

Original comment by `tstromb...@google.com` on 8 Dec 2009 at 12:39

**Comment by GoogleCodeExporter** [comment]
```
1.0.1 has been posted - can you confirm if this fixes the bug, or simply moves 
the 
crash around?

```

Original comment by `tstromb...@google.com` on 8 Dec 2009 at 2:00

**Comment by GoogleCodeExporter** [comment]
```
I do not get the same error now. However it still does not work. 
The unzip progress bar goes to the end and then freezes. It says "Unzipping... 
(Not
Responding)". Then the namebench menu comes up and when I click on "Start 
Benchmark"
I see a pop up that says:
list index out of range

File"libnamebench\nameserver_list.pyc", Line 123, in_init_

File"libnamebench\nameserver_list.pyc",line 209, in
ApplyCongestionFactor

File"libnamebench\conn_quality.pyc",line 50, in
CheckConnectionQuality

File"libnamebench\conn_quality.pyc",line 44, in
GetNegativeResponseDuration
```

Original comment by `alexwie...@gmail.com` on 8 Dec 2009 at 2:42

Attachments:
- [namebencherr1.jpg](https://storage.googleapis.com/google-code-attachments/namebench/issue-40/comment-4/namebencherr1.jpg)

**Comment by GoogleCodeExporter** [comment]
```
Also when I close out of the program I get this error

"See the Logfile at C:\Users\Alex\AppData\Local\Temp\ckz_6SPR\namebench.exe.log"

This is what the file says:

Traceback (most recent call last):
  File "libnamebench\tk.pyc", line 85, in run
  File "libnamebench\base_ui.pyc", line 58, in PrepareBenchmark
  File "libnamebench\nameserver_list.pyc", line 123, in __init__
  File "libnamebench\nameserver_list.pyc", line 209, in ApplyCongestionFactor
  File "libnamebench\conn_quality.pyc", line 50, in CheckConnectionQuality
  File "libnamebench\conn_quality.pyc", line 44, in GetNegativeResponseDuration
IndexError: list index out of range

```

Original comment by `alexwie...@gmail.com` on 8 Dec 2009 at 2:45

**Comment by GoogleCodeExporter** [comment]
```
This should be fixed in version 1.0.2.

```

Original comment by `tstromb...@google.com` on 8 Dec 2009 at 7:35
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Everything appears to be working now. Thanks for the fixes. 
```

Original comment by `alexwie...@gmail.com` on 9 Dec 2009 at 2:35
